### PR TITLE
feat(typescript): add `disableTsconfigDev` property to not generate the `tsconfig.dev.json`

### PR DESCRIPTION
**Pull request by gmukul01**
Allows us to not generate the `tsconfig.dev.json` file and use the `tsconfig.json` for all the tasks.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/typescript/typescript.ts [hunk 1]
```diff
@@ -264,24 +270,28 @@ export class TypeScriptProject extends NodeProject {
...
-          exclude: ["node_modules"],
-          compilerOptions: compilerOptionDefaults,
-        },
-        options.tsconfig,
-        options.tsconfigDev
-      )
-    );
+    if (options.disableTsconfigDev && !options.disableTsconfig) {
+      this.tsconfigDev = this.tsconfig!;
+    } else {
```
Comment (mrgrain) [inline src/typescript/typescript.ts:275]: So we are disabling it, but then we are saying it is tsconfig?! I'm confused.

**Review by mrgrain — CHANGES_REQUESTED** [review]
Great idea! Will need some test cases and questions answered.

Should we make sure that either tsconfig or tsconfigDev is available?
What happens with .projenrc.ts task when the dev config is gone?

PS: You can already do this by doing `disableTsConfig` and changing the name of the dev config. But that's a bad API.

**Comment by gmukul01** [comment]
Should we make sure that either tsconfig or tsconfigDev is available? 
Answer: Yes I have make sure about this and even if you have passed both `disableTsconfigDev` and `disableTsconfig`  then it will definitely create the `tsconfig.dev.json`.

What happens with .projenrc.ts task when the dev config is gone?
Answer: Since I am assigning the default `tsconfig.json` to the `this.tsconfigDev`, all the tasks will consider the `tsconfig.json` for all the typescript commands.

**Comment by gmukul01** [comment]
@mrgrain Added the test cases also. :)

File: test/typescript/typescript.test.ts [hunk 2]
```diff
@@ -237,3 +237,57 @@ test("upgrade task ignores pinned versions", () => {
...
+    const prj = new TypeScriptProject({
+      name: "test",
+      projenrcTs: true,
+      defaultReleaseBranch: "main",
+      disableTsconfig: true,
+      disableTsconfigDev: true,
+    });
+
+    const snapshot = synthSnapshot(prj);
+    expect(prj.tsconfigDev.fileName).toBe("tsconfig.dev.json");
```
Comment (mrgrain) [inline test/typescript/typescript.test.ts:286]:
```suggestion
    expect(prj.tsconfigDev.fileName).toBe("tsconfig.dev.json");
    expect(snapshot["tsconfig.dev.json"]).not.toBeUndefined();
```

File: test/typescript/typescript.test.ts [hunk 3]
```diff
@@ -237,3 +237,57 @@ test("upgrade task ignores pinned versions", () => {
...
+describe("tsconfigDev", () => {
+  test("uses tsconfig.dev.json by default", () => {
+    const prj = new TypeScriptProject({
+      name: "test",
+      projenrcTs: true,
+      defaultReleaseBranch: "main",
+    });
+
+    const snapshot = synthSnapshot(prj);
+    expect(prj.tsconfigDev.fileName).toBe("tsconfig.dev.json");
```
Comment (mrgrain) [inline test/typescript/typescript.test.ts:250]:
```suggestion
    expect(prj.tsconfigDev.fileName).toBe("tsconfig.dev.json");
    expect(snapshot["tsconfig.json"]).not.toBeUndefined();
    expect(snapshot["tsconfig.dev.json"]).not.toBeUndefined();
```

**Review by mrgrain — CHANGES_REQUESTED** [review]
Nice  once @gmukul01 and thanks for the test cases. I've added a couple small improvement suggestions to them.

Re `disableTsconfigDev` AND `disableTsconfig`. I think we should at least log a warning if both are disabled. But wondering if we should simply prohibit that and throw an error instead. That makes it at least explicit. What do you think? I'll have a look how we do handle this in similar situations.

Either way, the docstrings will need to mention that as well.

Edit: I had a look. We should simply throw an error in  this case. It's what we do elsewhere.
Example `throw new Error("Cannot specify both 'contents' and 'source' fields.");`

File: test/typescript/typescript.test.ts (hunk 3 above)
Comment (gmukul01) [reply test/typescript/typescript.test.ts:250]: Done

**Review by mrgrain — CHANGES_REQUESTED** [review]
Good to go. :shipit: 🚀

**Pull request merged**